consensus: racy dbft.OnTransaction

I faced with the following issue during mixed consensus bench run (1 Go node, 3 C# nodes and Go RPC node):

```
2020-12-18T15:53:03.303Z	INFO	received Commit	{"validator": 3}
2020-12-18T15:53:03.303Z	INFO	received Commit	{"validator": 0}
2020-12-18T15:53:03.303Z	INFO	approving block	{"height": 7, "hash": "4c02d52305a6a8981bd1598f0c3076d6de15a44a60ca692e189cd8a7249f175c", "tx_count": 0, "merkle": "d01a598d07b3035a7331a98667eb62e20508c6062c6995edbfcb1aeb3a3e240e", "prev": "b20bfc4982a429a63d6fa484cf8f8eed3008bcf04d62584840abadee57742420"}
2020-12-18T15:53:03.304Z	INFO	initializing dbft	{"height": 8, "view": 0, "index": 2, "role": "Backup"}
2020-12-18T15:53:04.017Z	INFO	blockchain persist completed	{"persistedBlocks": 1, "persistedKeys": 20, "headerHeight": 7, "blockHeight": 7, "took": "13.912701ms"}
2020-12-18T15:53:08.405Z	INFO	received PrepareRequest	{"validator": 0, "tx": 1967}
2020-12-18T15:53:08.418Z	INFO	missing tx	{"count": 1546}
2020-12-18T15:53:08.694Z	INFO	received PrepareResponse	{"validator": 1}
2020-12-18T15:53:08.817Z	INFO	received PrepareResponse	{"validator": 3}
2020-12-18T15:53:09.102Z	INFO	received Commit	{"validator": 3}
2020-12-18T15:53:09.152Z	INFO	received Commit	{"validator": 1}
2020-12-18T15:53:09.525Z	INFO	received Commit	{"validator": 0}
2020-12-18T15:53:10.149Z	INFO	sending PrepareResponse	{"height": 8, "view": 0}
2020-12-18T15:53:10.149Z	INFO	sending Commit	{"height": 8, "view": 0}
2020-12-18T15:53:10.149Z	INFO	approving block	{"height": 8, "hash": "79267f728ffdf15e1bf0b87da3987964adbe71116cc395b34aa78371c289d4c0", "tx_count": 1967, "merkle": "b2b109e701c7906bccca5cb1216aff01d42bfca64a4b112eef3ac512bd59cecb", "prev": "4c02d52305a6a8981bd1598f0c3076d6de15a44a60ca692e189cd8a7249f175c"}
2020-12-18T15:53:10.768Z	INFO	initializing dbft	{"height": 9, "view": 0, "index": 2, "role": "Backup"}
panic: runtime error: slice bounds out of range [:-1]
goroutine 37035 [running]:
github.com/nspcc-dev/dbft.(*DBFT).OnTransaction(0xc0002742c0, 0x105e1a0, 0xc0009b3960)
	github.com/nspcc-dev/dbft@v0.0.0-20201216144126-a4bc58feae87/dbft.go:158 +0x26a
github.com/nspcc-dev/neo-go/pkg/consensus.(*service).eventLoop(0xc0001ca480)
	github.com/nspcc-dev/neo-go/pkg/consensus/consensus.go:286 +0x282
created by github.com/nspcc-dev/neo-go/pkg/consensus.(*service).Start
	github.com/nspcc-dev/neo-go/pkg/consensus/consensus.go:236 +0x9b
```

I believe that the reason is in race condition inside the dbft.OnTransaction method, but not sure whether this situation is possible in the real network (not under the high load).

--- a/go.mod
+++ b/go.mod
@@ -11,7 +11,7 @@ require (
 	github.com/gorilla/websocket v1.4.2
 	github.com/hashicorp/golang-lru v0.5.4
 	github.com/mr-tron/base58 v1.1.2
-	github.com/nspcc-dev/dbft v0.0.0-20201216144126-a4bc58feae87
+	github.com/nspcc-dev/dbft v0.0.0-20201221101812-e13a1a1c3cb2
 	github.com/nspcc-dev/rfc6979 v0.2.0
 	github.com/pierrec/lz4 v2.5.2+incompatible
 	github.com/prometheus/client_golang v1.2.1

